### PR TITLE
Model standards Super Resolution Net

### PR DESCRIPTION
**Pull request by ktangsali**
# PhysicsNeMo Pull Request

## Description
Adopt the model standards. 

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: physicsnemo/models/srrn/super_res_net.py [hunk 1]
```diff
@@ -182,36 +200,47 @@ def __init__(
...
         output = self.subpixel_convolutional_blocks(
             output
-        )  # (N, n_channels, w * scaling factor, h * scaling factor)
-        output = self.conv_block3(
-            output
-        )  # (N, 3, w * scaling factor, h * scaling factor)
+        )  # (B, conv_layer_size, D*s, H*s, W*s)
+        output = self.conv_block3(output)  # (B, out_channels, D*s, H*s, W*s)
 
         return output
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply physicsnemo/models/srrn/super_res_net.py]: The model is JIT supported and that currently breaks with `torch.compile`. Supporting torch.compile needs a larger refactor which is out of scope for current PR.

File: physicsnemo/models/srrn/super_res_net.py [hunk 2]
```diff
@@ -117,16 +129,22 @@ def __init__(
         activation_fn: str = "prelu",
     ):
         super().__init__(meta=MetaData())
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply physicsnemo/models/srrn/super_res_net.py:132]: Jaxtyping and torchscript are not compatible.

File: physicsnemo/models/srrn/super_res_net.py [hunk 3]
```diff
@@ -182,36 +200,47 @@ def __init__(
...
-        output = self.conv_block3(
-            output
-        )  # (N, 3, w * scaling factor, h * scaling factor)
+        )  # (B, conv_layer_size, D*s, H*s, W*s)
+        output = self.conv_block3(output)  # (B, out_channels, D*s, H*s, W*s)
 
         return output
 
 
 class ConvolutionalBlock3d(nn.Module):
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply physicsnemo/models/srrn/super_res_net.py:233]: These blocks are purely for internal use and are not intended to be used by external users. Additionally, we want to keep this source code in one place (and not split it into multiple files) because of the MIT license header

File: physicsnemo/models/srrn/super_res_net.py [hunk 4]
```diff
@@ -182,36 +200,47 @@ def __init__(
         )
 
     def forward(self, in_vars: Tensor) -> Tensor:
```
Comment (coreyjadams) [inline physicsnemo/models/srrn/super_res_net.py]: Should this have jax typing hints?  I see the conversation with greptile about torch compile, but I understood that torch compile and type hints are just fine together.

File: physicsnemo/models/srrn/super_res_net.py [hunk 5]
```diff
@@ -312,37 +268,32 @@ def __init__(
...
         self.activation_fn = activation_fn
+
+        # Put together the convolutional block as a sequence of the layers
         self.conv_block = nn.Sequential(*layers)
 
-    def forward(
-        self,
-        input: Float[Tensor, "batch in_channels depth height width"],
-    ) -> Float[Tensor, "batch out_channels depth_out height_out width_out"]:
+    def forward(self, input: Tensor) -> Tensor:
```
Comment (coreyjadams) [inline physicsnemo/models/srrn/super_res_net.py]: type hints, etc?

File: physicsnemo/models/srrn/super_res_net.py [hunk 6]
```diff
@@ -423,58 +364,45 @@ class _SubPixelConvolutionalBlock3d(nn.Module):
...
+        # These additional channels are shuffled to form additional pixels,
+        #  upscaling each dimension by the scaling factor
+        self.pixel_shuffle = PixelShuffle3d(scaling_factor)
         self.prelu = nn.PReLU()
 
-    def forward(
-        self,
-        input: Float[Tensor, "batch channels depth height width"],
-    ) -> Float[Tensor, "batch channels depth_out height_out width_out"]:
+    def forward(self, input: Tensor) -> Tensor:
```
Comment (coreyjadams) [inline physicsnemo/models/srrn/super_res_net.py]: missing type hints.

File: physicsnemo/models/srrn/super_res_net.py [hunk 7]
```diff
@@ -423,58 +364,45 @@ class _SubPixelConvolutionalBlock3d(nn.Module):
...
         r"""Forward pass of sub-pixel convolutional block."""
-        # Increase channel count
```
Comment (coreyjadams) [inline physicsnemo/models/srrn/super_res_net.py]: Type hints

File: physicsnemo/models/srrn/super_res_net.py [hunk 8]
```diff
@@ -520,18 +446,10 @@ def __init__(
 
         self.conv_layers = nn.Sequential(*layers)
 
-    def forward(
-        self,
-        input: Float[Tensor, "batch channels depth height width"],
-    ) -> Float[Tensor, "batch channels depth height width"]:
+    def forward(self, input: Tensor) -> Tensor:
```
Comment (coreyjadams) [inline physicsnemo/models/srrn/super_res_net.py]: type hints here too?

Comment (ktangsali) [reply physicsnemo/models/srrn/super_res_net.py]: Fixed in: https://github.com/NVIDIA/physicsnemo/pull/1336/commits/2ed734d59f75f1a7633f484e06f76a5f88c51267

File: physicsnemo/models/srrn/super_res_net.py (hunk 7 above)
Comment (ktangsali) [reply physicsnemo/models/srrn/super_res_net.py]: Fixed in https://github.com/NVIDIA/physicsnemo/pull/1336/commits/2ed734d59f75f1a7633f484e06f76a5f88c51267

File: physicsnemo/models/srrn/super_res_net.py (hunk 6 above)
Comment (ktangsali) [reply physicsnemo/models/srrn/super_res_net.py]: Fixed in https://github.com/NVIDIA/physicsnemo/pull/1336/commits/2ed734d59f75f1a7633f484e06f76a5f88c51267

File: physicsnemo/models/srrn/super_res_net.py (hunk 5 above)
Comment (ktangsali) [reply physicsnemo/models/srrn/super_res_net.py]: Fixed in https://github.com/NVIDIA/physicsnemo/pull/1336/commits/2ed734d59f75f1a7633f484e06f76a5f88c51267

File: physicsnemo/models/srrn/super_res_net.py (hunk 4 above)
Comment (ktangsali) [reply physicsnemo/models/srrn/super_res_net.py]: Fixed in https://github.com/NVIDIA/physicsnemo/pull/1336/commits/2ed734d59f75f1a7633f484e06f76a5f88c51267

Issue was with torscript + jaxtyping. Torchscript has been disabled in favor of torch compile now.
